src/test/java/datawave/query/util/AllFieldMetadataHelperTest.java

```diff
@@ -0,0 +1,440 @@
+     */
+    @BeforeEach
+    void setUp() throws AccumuloSecurityException, AccumuloException, TableExistsException {
+        accumuloClient = new InMemoryAccumuloClient("root", new InMemoryInstance(AllFieldMetadataHelper.class.toString()));
+        if (!accumuloClient.tableOperations().exists(TABLE_METADATA)) {
+            accumuloClient.tableOperations().create(TABLE_METADATA);
+        }
+        final Set<Authorizations> allMetadataAuths = Collections.emptySet();
+        final Set<Authorizations> auths = Collections.singleton(new Authorizations(AUTHS));
+        TypeMetadataHelper typeMetadataHelper = new TypeMetadataHelper(Maps.newHashMap(), allMetadataAuths, accumuloClient, TABLE_METADATA, auths, false);
+        CompositeMetadataHelper compositeMetadataHelper = new CompositeMetadataHelper(accumuloClient, TABLE_METADATA, auths);
+        helper = new AllFieldMetadataHelper(typeMetadataHelper, compositeMetadataHelper, accumuloClient, TABLE_METADATA, auths, allMetadataAuths);
+    }
+    
+    /**
+     * Clear the metadata table after each test.
+     */
+    @AfterEach
+    void tearDown() throws AccumuloException, TableNotFoundException, AccumuloSecurityException {
+        accumuloClient.tableOperations().deleteRows(TABLE_METADATA, null, null);
+    }
+    
+    /**
+     * Write the given mutations to the metadata table.
+     */
+    private void writeMutations(Collection<Mutation> mutations) {
+        BatchWriterConfig config = new BatchWriterConfig();
+        config.setMaxMemory(0);
+        try (BatchWriter writer = accumuloClient.createBatchWriter(TABLE_METADATA, config)) {
+            writer.addMutations(mutations);
+            writer.flush();
+        } catch (MutationsRejectedException | TableNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Tests for {@link AllFieldMetadataHelper#getFieldIndexHoles()}.
+     */
+    @SuppressWarnings("unchecked")
+    @Nested
+    public class FieldIndexHoleTests {
+        
+        private final Supplier<Map<String,Map<String,FieldIndexHole>>> INDEX_FUNCTION = () -> {
+            try {
+                return helper.getFieldIndexHoles();
+            } catch (TableNotFoundException | CharacterCodingException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        
+        private final Supplier<Map<String,Map<String,FieldIndexHole>>> REVERSED_INDEX_FUNCTION = () -> {
+            try {
+                return helper.getReversedFieldIndexHoles();
+            } catch (TableNotFoundException | CharacterCodingException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        
+        private Supplier<Map<String,Map<String,FieldIndexHole>>> getIndexHoleFunction(String cf) {
+            return cf.equals("i") ? INDEX_FUNCTION : REVERSED_INDEX_FUNCTION;
+        }
+        
+        /**
+         * Test against data that has no field index holes.
+         */
```

Comment:
Need a test where we have many fields in a row that only have 'f' entries and no 'i' entries.

Reply:
A test like that is already present in `testAllDatesAreIndexHoles()` [here](https://github.com/NationalSecurityAgency/datawave-metadata-utils/pull/29/files/f2bc5377d7548f9f18283852ff42727a398efd53#diff-da568db7f6796bd73bbb2e23b7124f99a8feb9ce24dbfff670a54fbeb5bc1e13R306), where multiple 'f' entries are created for different fields in a row with no 'i' entries.